pkg/dfns-api-client/auth_transport.go

```diff
@@ -0,0 +1,286 @@
+package dfns_api_client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+
+	"github/dfns/dfns-sdk-go/pkg/credentials"
+)
+
+const (
+	UserActionHeader    = "x-dfns-useraction"
+	appIDHeader         = "x-dfns-appid"
+	nonceHeader         = "x-dfns-nonce"
+	appSecretHeader     = "x-dfns-appsecret" //nolint:gosec
+	authorizationHeader = "authorization"
+	contentTypeHeader   = "Content-Type"
+)
+
+type authTransport struct {
+	*DfnsBaseApiOptions
+}
+
+func (auth *authTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	auth.setHeaders(req)
+
+	methodsWithAuth := map[string]struct{}{
+		"POST":   {},
+		"PUT":    {},
+		"DELETE": {},
+	}
```

Comment:
why a map not an array?

Reply:
faster to check if a key is among these values

Reply:
Maybe a bit over-optimized imo, given time to execeute a list search & frequency of calling this 😄  It gives up a bit on readability, but it works